### PR TITLE
Fix CMakeLists + package.xmls

### LINKED ISSUE #537: Indigo build fails when ROS java is installed
Several packages fail to build. For example, **navfn** fails:

```bash
Traceback (most recent call last):
  File "/opt/ros/indigo/share/genjava/cmake/../../../lib/genjava/genjava_gradle_project.py", line 14, in <module>
    genjava.main(sys.argv)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/genjava/genjava_main.py", line 82, in main
    gradle_project.create(args.package, args.output_dir)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/genjava/gradle_project.py", line 152, in create
    raise IOError("could not find %s on the ros package path" % msg_pkg_name)
IOError: could not find navfn on the ros package path
```

and **robot\_pose\_ekf**:

```bash
Traceback (most recent call last):
  File "/opt/ros/indigo/share/genjava/cmake/../../../lib/genjava/genjava_gradle_project.py", line 14, in <module>
    genjava.main(sys.argv)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/genjava/genjava_main.py", line 82, in main
    gradle_project.create(args.package, args.output_dir)
  File "/opt/ros/indigo/lib/python2.7/dist-packages/genjava/gradle_project.py", line 152, in create
    raise IOError("could not find %s on the ros package path" % msg_pkg_name)
IOError: could not find robot_pose_ekf on the ros package path
```

But the build succeeds upon removal of ROS java ([Ref](http://answers.ros.org/question/213859/failed-to-build-razor_imu_9dof/?answer=222562#post-id-222562)) with the following command:

```bash
sudo apt-get remove ros-indigo-genjava
```

### PR DESCRIPTION
**Pull request by mintar**
This fixes compilation errors when rosjava is installed on the system.
It also removes a lot of errors reported by catkin_lint.

Fixes #537 .

### CODE REVIEW

**Comment by spmaniato** [comment]
@mintar Were there no complaints about `move_base` ? See [this comment](https://github.com/ros-planning/navigation/issues/483#issuecomment-235446009) in #483

**Comment by mintar** [comment]
@spmaniato I didn't even run `catkin_lint` on `move_base`, since with this PR it compiles for me with `rosjava` installed. But one should definitely go through all the packages again and fix all `catkin_lint` errors.

Some more remarks (they may be obvious, but nevertheless):

* The bug #537 only occurs when `rosjava` is installed *and* `navigation` is checked out as a source package into the Catkin workspace you're building.
* Even after applying this PR, I initially need to run `catkin_make` 2-3 times before it eventually succeeds due to some problems with `dynamic_reconfigure`. (Without this PR, it doesn't build at all.)

**Comment by mintar** [comment]
Ping

**Comment by caicaixia** [comment]
Hello,  these changes haven't released yet, right? I just upgrade the ros-indigo-navigation, but the version is: 1.12.13-0trusty-20161128-101040-0800.  And still have ros java problem :(

**Comment by mintar** [comment]
@caicaixia : No, I'm still waiting for this PR to be merged. But this bug only concerns source installs anyway, so it doesn't matter if this is released or not.

File: costmap_2d/CMakeLists.txt [hunk 1]
```diff
@@ -60,8 +61,6 @@ generate_dynamic_reconfigure_options(
 catkin_package(
     INCLUDE_DIRS
         include
-        ${EIGEN_INCLUDE_DIRS}
-        ${PCL_INCLUDE_DIRS}
```
Comment (DLu) [inline costmap_2d/CMakeLists.txt:64]: Why are these removed?

**Review by DLu — COMMENTED** [review]
Overall, looks good. I merged things in the wrong order (see #549), so could you please merge the latest indigo-devel?

File: costmap_2d/CMakeLists.txt (hunk 1 above)
Comment (mintar) [reply costmap_2d/CMakeLists.txt:64]: Because there already is a catkin_package(DEPENDS PCL Eigen). This means that PCL + Eigen are already added as dependencies for any project that depends on `costmap_2d`, and it's Eigen's (resp. PCL's) job to export their own include dirs. The `catkin_package(INCLUDE_DIRS)` should only export our own include dirs.

Removing these two lines makes the following `catkin_lint` error go away:
```
costmap_2d: CMakeLists.txt(61): error: catkin_package() exports non-package include path
     * You listed one or more include paths in the INCLUDE_DIRS stanza
     * of your catkin_package() call which are not part of your
     * package. If you want to export include paths of other modules,
     * use find_package(), find_path(), and/or find_library() and add
     * the dependency to the DEPENDS stanza.
```

**Comment by mintar** [comment]
I've rebased this branch onto the latest indigo-devel, so it should be good to merge.

**Comment by mintar** [comment]
I've just added two more commits that fix 87 out of 90 `catkin_lint` errors (the rest are false positives).

**Comment by mikeferguson** [comment]
@mintar -- could you rebase this on top of latest changes (there is a conflict) and then we can get this merged. Thanks!

**Comment by mintar** [comment]
@mikeferguson Ok, rebased!

**Comment by mintar** [comment]
ping

**Comment by mikeferguson** [comment]
Test failures are unrelated. I'm going to squash-merge to make it easier to forward port to K+L

**Comment by mintar** [comment]
Hooray! :-)
